src/errorhandler/errorhandler.cpp

```diff
@@ -0,0 +1,87 @@
+#include "errorhandler.hpp"
+
+sErrorStructure cErrorHandler::ErrorStructure[E_NUMBER_OF_ERRORS] =
+{
+// Error number         |       Error active        |       Error enable        |       Reaction
+{  E_FREE_TO_USE,              false,                      E_ERROR_ENABLED,            NONE_REACT}
```

Comment:
"Error number" specifies the error you are configuring, "Error enable" and "Reaction" is part of configuration and "Error active" is internal state (flag).
I would seperate configuration and internal state.
So you would have array of errors - that would have internal state (is error active?, is error already in error table?, ...), and configuration that could be fully in flash.